pkg/disttask/framework/taskexecutor/task_executor_test.go

```diff
@@ -546,3 +549,33 @@ func TestInject(t *testing.T) {
 	got := e.GetResource()
 	require.Equal(t, r, got)
 }
+
+func callOnError(taskExecutor *BaseTaskExecutor) {
+	taskExecutor.onError(errors.Errorf("mock error"))
+}
+
+func TestExecutorOnErrorLog(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+	mockSubtaskTable := mock.NewMockTaskTable(ctrl)
+	mockExtension := mock.NewMockExtension(ctrl)
+
+	ctx := context.Background()
+	task := &proto.Task{TaskBase: proto.TaskBase{Step: proto.StepOne, Type: "type", ID: 1, Concurrency: 1}}
+	taskExecutor := NewBaseTaskExecutor(ctx, "tidb1", task, mockSubtaskTable)
+	taskExecutor.Extension = mockExtension
+
+	observedZapCore, observedLogs := observer.New(zap.ErrorLevel)
+	observedLogger := zap.New(observedZapCore)
+	taskExecutor.logger = observedLogger
+
+	callOnError(taskExecutor)
+	require.GreaterOrEqual(t, observedLogs.Len(), 1)
+	errLog := observedLogs.All()[0]
+	stack := errLog.ContextMap()["stack"]
+	require.IsType(t, "", stack)
+	stackStr := stack.(string)
+	require.Truef(t, strings.HasPrefix(stackStr,
```

Comment:
Hi, the problem is if the error that passed to `onError` lost its trace. You can reproduce it like

```golang
func throwError() error {
	return errors.New("mock error")
}

func callOnError(taskExecutor *BaseTaskExecutor) {
	taskExecutor.onError(errors.Trace(throwError()))
}
```

the first line on stack should be `throwError` function

Reply:
Hi @lance6716 ,

Please correct me if I misunderstand.

In my knowledge, the order of popping stack frame should be `throwError` -> `errors.Trace` -> `taskExecutor.onError`.

So, when the `errors.Trace` is invoked, the previous stack info has been popped and can't be retrieved again.

The only way is explicitly adding `errors.Trace` before popping `throwError`.

I have added the `errors.Trace` before all `onError`.

Best Regards,
Edward

Reply:
Got it!

Thanks very much for your help and suggestions! 😊

I will check the `DisableErrorVerbose` option and the stack inside the error itself.

Best Regards,
Edward

Reply:
Just to remind that the user may turn off `DisableErrorVerbose` in config. We should find another way to print error stack.

Reply:
Hi @lance6716 ,

I think the stable way to print the top stack should be the error field itself.

Since the stack field works now, I will try to embed the top stack info inside the error message itself.

Best Regards,
Edward